waspc/data/Generator/templates/react-app/src/index.js

```diff
@@ -4,22 +4,26 @@ import ReactDOM from 'react-dom'
 import { QueryClientProvider } from '@tanstack/react-query'
 
 import router from './router'
-import { 
+import {
   initializeQueryClient,
   queryClientInitialized,
 } from './queryClient'
 import * as serviceWorker from './serviceWorker'
 
-{=# doesClientSetupFnExist =}
-{=& clientSetupJsFnImportStatement =}
-{=/ doesClientSetupFnExist =}
+{=# setupFn.isDefined =}
+{=& setupFn.importStatement =}
+{=/ setupFn.isDefined =}
+
+{=# rootComponent.isDefined =}
+{=& rootComponent.importStatement =}
+{=/ rootComponent.isDefined =}
 
 startApp()
 
 async function startApp() {
-  {=# doesClientSetupFnExist =}
-  await {= clientSetupJsFnIdentifier =}()
-  {=/ doesClientSetupFnExist =}
+  {=# setupFn.isDefined =}
```

Comment:
Ideally the check would be just `{=# setupFn =}`, but Mustache doesn't trigger `{=^ setupFn =}` when it's `undefined` vs. `false`, which is weird. So, that's why we use this `isDefined` crutch to make it work in all cases.